doc/rst/technotes/libraries.rst

```diff
@@ -111,23 +205,145 @@ example looks like this:
+program that will use the library):
 
 .. code-block:: sh
 
-   `$CHPL_HOME/util/config/compileline --compile` myprog.c -L. -lfoo `$CHPL_HOME/util/config/compileline --libraries`
+   `$CHPL_HOME/util/config/compileline --compile` myCProg.c -Llib/ -lfoo `$CHPL_HOME/util/config/compileline --libraries`
 
 Note that ``compileline --compile-c++`` is also available. It requests a
 compilation command able to compile a C++ program.
 
+Using Your Library in Python
+============================
+
+Pre-requisites
+--------------
+
+To make use of your library in Python with minimal work, the Chapel compiler
+requires that you have ``python3`` installed in your ``$PATH``, and that your
+``python3`` installation contains ``numpy`` and ``Cython``.
+
+We don't anticipate supporting ``python2`` at this time, sorry.
+
+Compiling Your Chapel Library
+-----------------------------
+
+In addition to compiling with ``--library``, it is recommended to compile with
+``--library-python``.  This will create a Python-compatible module in addition
+to the normally generated library and header file.
+
+Python Module Name
+------------------
+
+By default, the name of the generated Python module will match the base name
+of the generated library.  In the ``foo.chpl`` example, this would mean the
+Python module will be named ``foo``.
+
+The name of the generated module can be changed independently of the generated
+library name using the compilation flag ``--library-python-name``.
+
+PYTHONPATH
+----------
+
+To use your library in a Python program, you will need to extend your
+``PYTHONPATH`` environment variable to include the directory where your library
+files are generated, e.g.:
+
+.. code-block:: sh
+
+   export PYTHONPATH=lib/:$PYTHONPATH
+
+See :ref:`Location of the Generated Library` for where your library files are
+generated, and how to change this location when compiling your Chapel library.
+
+Initializing and Using Your Library in Python
+---------------------------------------------
+
+Once your ``PYTHONPATH`` is set up and the Python module created, you can
+``import`` the module like a normal Python module.
+
+Similarly to using your library with C, you will need to call a set up function
+to ensure the Chapel runtime and standard modules are initialized, as well as
+a clean up function.
+
+Unlike the C case, the set up function is called ``chpl_setup()`` and will also
+handle initializing your module.   This function will still need to be called
+prior to any Chapel library function calls.
+
+Also unlike the C case, the clean up function is called ``chpl_cleanup()``.
+This function will still need to be called after all the Chapel library function
+calls are finished.
+
+For example:
+
+.. code-block:: Python
+
+   import foo
+
+   foo.chpl_setup()
+
+   foo.baz(7) // Call into a library function
+
+   foo.chpl_cleanup()
+
+.. note::
+
+   Currently the ``chpl_cleanup()`` function will also cause the Python program
```

Comment:
The ``chpl_cleanup()`` function will also cause the Python program  to exit. Make sure your
Python functionality is also complete before calling this function.